app/models/spree/line_item.rb

```diff
@@ -222,6 +222,15 @@ def scoper
       @scoper ||= OpenFoodNetwork::ScopeVariantToHub.new(order.distributor)
     end
 
+    def sort_for_invoice
+      if order.distributor.preferred_invoice_order_by_supplier
+        line_items = order.line_items.managed_by(supplier)
+        line_items.sort_by { |li| [li.supplier.name, li.product.name] }
+      else
+        line_items.sort_by { |li| [li.product.name] }
+      end
+    end
+
```

Comment:
This method is currently not used. And it doesn't look quite right but I like the `sort_by` use.

Maybe you can define it on the Order model.

Reply:
I'll give it a go and see what happens 👋